curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64caf237baef43587be6d860.md

```diff
@@ -0,0 +1,356 @@
+---
+id: 64caf237baef43587be6d860
+title: Step 87
+challengeType: 0
+dashedName: step-87
+---
+
+# --description--
+
+Add an `if` statement that checks if every platform detection rule is `true`. Make sure to use the `every` method for this.
+
+# --hints--
+
+You should have an `if` statement that uses the `every` method to check if every platform detection rule is `true`. 
+
+```js
+assert.match(code, /\s*if\s*\(platformDetectionRules\.every\(\((.+)\)\s*=>\s*\1\s*\)\s*\)\s*{\s*}\s*;?/);
```

Comment:
```suggestion
assert.match(code, /if\s*\(platformDetectionRules\.every\(rule\s*=>\s*rule\)\)\s*\{\s*\};?/);
```

Reply:
I don't think that having `rule` explicitly written in the regex is a good choice. The word "rule" from the instructions will be translated, and the campers will not be able to guess the parameter name.

Reply:
What do you suggest we do then?

Reply:
Currently test will accept any parameter name.

It came to my mind now, that this can be done simpler. Instead of expecting arrow function, ask camper to use `Boolean` as a callback for `.every` method.

```suggestion
assert.match(code, /\s*if\s*\(platformDetectionRules\.every\(\s*Boolean\s*\)\s*{\s*}\s*;?/);
```

Reply:
I have read through all of the comments here and I propose we leave this test as is where the user can pass in any parameter name they want.
That will be consistent with the other times the `every` method not only comes up in this project but in other projects like the spreadsheet project 👍